src/components/ui.js

```diff
@@ -550,27 +555,14 @@ AFRAME.registerComponent('ui', {
       material.needsUpdate = true;
     });
 
+    this.el.setAttribute('animation__showmessage', { dur: 500, property: 'ui.opacity', from: 0, to: 1, startEvents: 'showmessage' });
+    this.el.setAttribute('animation__hidemessage', { dur: 500, delay: 3000, property: 'ui.opacity', from: 1, to: 0, startEvents: 'hidemessage' });
+    this.el.addEventListener('animationcomplete__showmessage', function (evt) {
+      self.el.emit('hidemessage');
+    });
     function showMessage (msgObject) {
       msgObject.visible = true;
-      var object = { opacity: 0.0 };
-      var tween = new AFRAME.TWEEN.Tween(object)
-        .to({opacity: 1.0}, 500)
-        .onUpdate(function () {
-          self.messagesMaterial.opacity = object.opacity;
-        })
-        .chain(
-          new AFRAME.TWEEN.Tween(object)
-            .to({opacity: 0.0}, 500)
-            .delay(3000)
-            .onComplete(function () {
-              msgObject.visible = false;
-            })
-            .onUpdate(function () {
-              self.messagesMaterial.opacity = object.opacity;
-            })
-          );
-
-      tween.start();
+      self.el.emit('showmessage');
```

Comment:
Same for this one we can perhaps hook up the animation component to one of the emitted events from the animations.